sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/sql/execute/result/StreamQueryResult.java

```diff
@@ -68,12 +68,12 @@ public boolean next() throws SQLException {
     
     @Override
     public Object getValue(final int columnIndex, final Class<?> type) throws SQLException {
-        return decrypt(columnIndex, QueryResultUtil.getValue(resultSet, columnIndex));
+        return decrypt(columnIndex, QueryResultUtil.getValue(resultSet, columnIndex, type));
```

Comment:
StreamQueryResult pass the type to getValue(), but what about MemoryQueryResult?

Reply:
> StreamQueryResult pass the type to getValue(), but what about MemoryQueryResult?

This is why i doesn't deleted `QueryResultUtil.getValueByColumnType()`, I have overload a `getValue()` method with `Class<?> type` in `QueryResultUtil`,  `StreamQueryResult` will call the new `getValue` with `Class<?> type`, and `MemoryQueryResult ` will continue to call the old `getValue()`.
![image](https://user-images.githubusercontent.com/14866067/60784761-c685f900-a182-11e9-8ef7-31371658c823.png)
![image](https://user-images.githubusercontent.com/14866067/60784778-d1408e00-a182-11e9-8229-452a004fb6e0.png)

Reply:
> StreamQueryResult pass the type to getValue(), but what about MemoryQueryResult?

This is why I doesn't deleted `QueryResultUtil.getValueByColumnType()`, I just have overload a `getValue()` method with `Class<?> type` in `QueryResultUtil`,  `StreamQueryResult` will call the new `getValue()` with `Class<?> type`, and `MemoryQueryResult ` will continue to call the old `getValue()`.
![StreamQueryResult call the new getValue](https://user-images.githubusercontent.com/14866067/60784778-d1408e00-a182-11e9-8229-452a004fb6e0.png)
![MemoryQueryResult call the old getValue](https://user-images.githubusercontent.com/14866067/60784761-c685f900-a182-11e9-8ef7-31371658c823.png)

Reply:
If so, the problem was not resolved. Two modes get two different value is even worse, isn't?

Reply:
It's my fault, I overlooked this.
And I have a question. Why do we need to create a QueryRow Iterator and manage it ? Why not do just like [druid's DruidPooledResultSet](https://github.com/alibaba/druid/blob/master/src/main/java/com/alibaba/druid/pool/DruidPooledResultSet.java) ？

Reply:
It's OK. We have noticed your work and expect more contribution from you. I suggest you become a contributor first, then try to become a committer.

> And I have a question. Why do we need to create a QueryRow Iterator in MemoryQueryResult and manage it ? Why not do just like [druid's DruidPooledResultSet](https://github.com/alibaba/druid/blob/master/src/main/java/com/alibaba/druid/pool/DruidPooledResultSet.java) ？

Sometimes we need to get all data to memory:
1. Merge engine:
https://shardingsphere.apache.org/document/current/en/features/sharding/principle/merge/
2. Execute engine:
https://shardingsphere.apache.org/document/current/en/features/sharding/principle/execute/

Reply:
I'll read it first, thank you.